pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -43,6 +43,7 @@ const (
 	clusterPoolAdminRoleName        = "hive-cluster-pool-admin"
 	clusterPoolAdminRoleBindingName = "hive-cluster-pool-admin-binding"
 	icSecretDependent               = "install config template secret"
+	cdClusterPoolIndex              = "spec.clusterpool.namespacedname"
```

Comment:
if possible we should try to mimic the actual path of the value or as close as we can be. so my personal preference would be `spec.clusterPoolRef`

Reply:
I've noticed other indices don't mirror actual paths exactly.

https://github.com/openshift/hive/blob/33be31582aa140ce772333d0437eb77cd723cee2/pkg/controller/machinemanagement/machinemanagement_controller.go#L88
https://github.com/openshift/hive/blob/33be31582aa140ce772333d0437eb77cd723cee2/pkg/controller/clusterdeployment/clusterinstalls.go#L223
https://github.com/openshift/hive/blob/33be31582aa140ce772333d0437eb77cd723cee2/pkg/operator/hive/hive_controller.go#L160

I don't know if there's a technical reason -- like avoiding actual conflicts somewhere in c-r/apiserver -- but
- It actually helped me when figuring out how to do this, because it made it clear that we were *not* referencing/indexing exact fields; and
- In this case we're not indexing on the whole ClusterPoolRef. In particular, we're not using the ClaimName.

I don't mind changing the string if you have another suggestion, but I would like it *not* to be `spec.clusterPoolRef`.